dataland-frontend/src/components/resources/dataRequest/ReviewRequestButtons.vue

```diff
@@ -0,0 +1,259 @@
+        :answered-data-requests="answeredDataRequestsForViewPage"
+        :action-on-click="actionOnClick"
+        @selected-reporting-period="handleReportingPeriodSelection"
+      />
+    </OverlayPanel>
+  </div>
+</template>
+
+<script lang="ts">
+import PrimeButton from "primevue/button";
+import { defineComponent, inject } from "vue";
+import type Keycloak from "keycloak-js";
+import { getAnsweredDataRequestsForViewPage, patchDataRequestStatus } from "@/utils/RequestUtils";
+import OverlayPanel from "primevue/overlaypanel";
+import { type DataMetaInformation, type DataTypeEnum, type ErrorResponse } from "@clients/backend";
+import SelectReportingPeriodDialog from "@/components/general/SelectReportingPeriodDialog.vue";
+import { ReportingPeriodTableActions, type ReportingPeriodTableEntry } from "@/utils/PremadeDropdownDatasets";
+import { RequestStatus, type StoredDataRequest } from "@clients/communitymanager";
+import PrimeDialog from "primevue/dialog";
+import { assertDefined } from "@/utils/TypeScriptUtils";
+import { AxiosError } from "axios";
+
+export default defineComponent({
+  name: "ReviewRequestButtons",
+  setup() {
+    return {
+      getKeycloakPromise: inject<() => Promise<Keycloak>>("getKeycloakPromise"),
+    };
+  },
+  components: { PrimeButton, OverlayPanel, SelectReportingPeriodDialog, PrimeDialog },
+  props: {
+    companyId: {
+      type: String,
+      required: true,
+    },
+    framework: {
+      type: String,
+      required: true,
+    },
+    mapOfReportingPeriodToActiveDataset: {
+      type: Map,
+      required: true,
+    },
+  },
+  emits: ["isVisible"],
+  computed: {
+    isVisible() {
+      return this.answeredDataRequestsForViewPage.length > 0;
+    },
+  },
+  watch: {
+    isVisible(newStatus: boolean) {
+      this.$emit("isVisible", newStatus);
+    },
+  },
+  data() {
+    return {
+      answeredDataRequestsForViewPage: [] as StoredDataRequest[],
+      dialogIsVisible: false,
+      dialog: "Default\n text.",
+      dialogIsSuccess: false,
+      actionOnClick: ReportingPeriodTableActions.ReopenRequest,
+    };
+  },
+  mounted() {
+    void this.updateAnsweredDataRequestsForViewPage();
+  },
+  methods: {
+    /**
+     * Closes the SuccessModal
+     */
+    closeSuccessModal() {
+      this.dialogIsVisible = false;
+    },
+    /**
+     * Opens the SuccessModal with given dialog
+     * @param dialog desired dialog
+     * @param dialogIsSuccess if false, display error message
+     */
+    openSuccessModal(dialog: string, dialogIsSuccess: boolean = true) {
+      this.dialogIsSuccess = dialogIsSuccess;
+      this.dialog = dialog;
+      this.dialogIsVisible = true;
+    },
+    /**
+     * Makes the api call and updates the list of answered data requests.
+     */
+    async updateAnsweredDataRequestsForViewPage() {
+      this.answeredDataRequestsForViewPage = await getAnsweredDataRequestsForViewPage(
+        this.companyId,
+        this.framework as DataTypeEnum,
```

Comment:
is it possible/practical to define framework as DataTypeEnum instead of converting it?

Reply:
let me check..

Reply:
yes, worked. I changed it and could even get rid of one condition and slim the test to it